cli/azd/pkg/project/framework_service.go

```diff
@@ -79,22 +78,25 @@ type FrameworkService interface {
 	Restore(
 		ctx context.Context,
 		serviceConfig *ServiceConfig,
-	) *async.TaskWithProgress[*ServiceRestoreResult, ServiceProgress]
+		logProgress LogProgressFunc,
```

Comment:
Last we discussed in our engineering forum we agreed that we would prefer a solution where we don't need to modify the parameters of our methods and instead prefer a solution where we could inject some progress reporter via dependency injection.